src/java/org/apache/cassandra/cql3/statements/SelectStatement.java

```diff
@@ -1127,27 +1127,26 @@ private boolean needIndexOrdering()
     /**
      * Orders results when multiple keys are selected (using IN)
      */
-    public void orderResults(ResultSet cqlRows, QueryOptions options)
+    public SortedRowsBuilder sortedRowsBuilder(int limit, int offset, QueryOptions options)
     {
-        if (cqlRows.size() == 0 || !needsPostQueryOrdering())
-            return;
-
-        if (orderingComparator != null)
+        if (!needsPostQueryOrdering() || orderingComparator == null)
```

Comment:
Is it possible to have:
*  `needsPostQueryOrdering == true` and `orderingComparator == null`?
*  `needsPostQueryOrdering == false` and `orderingComparator != null`?

If not, then we don't need the `||` part. 
You can add assert to document the valid / invalid combinations.